website_lazy_load_image/models/ir_ui_view.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2018 Onestein
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+from lxml import etree
+
+
+class IrUiView(models.Model):
+    _inherit = 'ir.ui.view'
+
+    DEFAULT_SRC = 'data:image/gif;base64,R0lGODlhAQABAIAAAP///' \
+                  'wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw=='
+
+    @api.model
+    def render_template(self, template, values=None, engine='ir.qweb'):
+        res = super(IrUiView, self).render_template(template, values, engine)
+        website_id = self.env.context.get('website_id')
+        if website_id and not \
+                self.env['website'].browse(website_id).is_publisher():
+            html = etree.HTML(res)
+            imgs = html.xpath('//main//img[@src]') + html.xpath(
```

Comment:
I'd exclude images having already a class "lazy" and I'd handle background-image styles if any.
The latter can be postponed to a ROADMAP item if you don't have time